dist/logic.js

```diff
@@ -0,0 +1,116 @@
+class Model {
+    constructor() {
+        this.cityData = []
+    }
+
+    momentDateFormat (city){
+       return moment(city.updatedAt).format('LLLL')
+    }
+
+    async getDataFromDB() {
+        const fatch = await $.get("/cities")
+        fatch.forEach(a => {
+            a.updatedAt = moment(a.updatedAt).format('LLLL')
+            this.cityData.push(a)
+            console.log(a)
+        });
+    }
+   
+    async getCityData(cityName, lat, lon) {
+        if (!cityName) {
+            return alert ("try again")
+        }
+        let exist = false
+        this.cityData.forEach(a => {
+            if (cityName === a.name.toLowerCase()) {
+                exist = true
+            }
+        });
+        if (exist) {
+            return alert("City already exist, you can update your data")
+        }
+        if (lat && lon) {
+            const fatch = await $.get(`/city/${cityName}?lat=${lat}&lon=${lon}`)
+            this.cityData.push({ ...fatch, saved: false })
+            console.log(fatch)
+        }
+        else{
+            const fatch = await $.get(`/city/${cityName}`)
+            this.cityData.push({ ...fatch, saved: false })
+            console.log(fatch)
+        }
+    }
+
+
+    async getPosition() {
+        return new Promise(function (resolve, reject) {
+            if(navigator.geolocation){
+                navigator.geolocation.getCurrentPosition(resolve, reject);
+            }
+        });
+    }
+
+    async getCurrentCity() {
+        let position = await this.getPosition()
+        await this.getCityData("dummyCity", position.coords.latitude, position.coords.longitude)
```

Comment:
At the beggining of  getCityData function (line 18) i add 
if(!cityName) for times I'm getting undifined or null should i remove this if statement? (I mainly add this to alert the user when he press search without typing anything).

Reply:
So then this should be in the controller. That way if it is `null`, instead of `alert`ing something to the user, you can communicate with the `renderer` to render some sort of message to the user.